### PR TITLE
chore: fix a bug in cart-item-list.component.ts

### LINKED ISSUE #15286: Input of item counter in SavedCartDetailsItems is disabled
**Describe the bug**
The middle section input of the item-counter in Saved Cart Details is disabled, but the quantity can be incremented/decremented with [-] [+] buttons.

**To Reproduce**
1. Create a shopping cart and save cart
2. Navigate to 'Saved Carts' under 'My Account'
3. Click on the saved cart
4. See error on item counter

**Expected behavior**
Uniform behaviour of item-counter  across features

**Screenshots**
Screen Shot 2022-03-02 at 4 07 40 PM

### PR DESCRIPTION
**Pull request by WeizhengSap**
close #15286

### CODE REVIEW

File: feature-libs/cart/base/components/cart-shared/cart-item-list/cart-item-list.component.ts [hunk 1]
```diff
@@ -168,12 +168,18 @@ export class CartItemListComponent implements OnInit, OnDestroy {
...
-      const group = new FormGroup({
-        entryNumber: new FormControl(item.entryNumber),
-        quantity: new FormControl(item.quantity, { updateOn: 'blur' }),
-      });
-
-      this.form.setControl(controlName, group);
+      const control = this.form.get(controlName);
+      if (control) {
+        if (control.get('quantity')?.value !== item.quantity) {
+          control.patchValue({ quantity: item.quantity }, { emitEvent: false });
```
Comment (bgambocjaviniar) [inline feature-libs/cart/base/components/cart-shared/cart-item-list/cart-item-list.component.ts:174]: In what case do we ever land in here? I tried smoke testing this, but I can't find an instance on where this fires, except the unit tests due to direct manipulation of items.

However, I understand if we have an existing control, then let's just give the value to what the item currently have as quantity. But to me it's the same if we remove the if else and just create a new control, but again, this never happens on the app, but only in the unit tests

------------
Basically, I favor keeping what it was before, but changing 

this.form.setControl(controlName, group);

to 

this.form.addControl(controlName, group);
**as it was before**

File: feature-libs/cart/base/components/cart-shared/cart-item-list/cart-item-list.component.spec.ts [hunk 2]
```diff
@@ -297,6 +297,17 @@ describe('CartItemListComponent', () => {
...
 
+    it('should update controls when quantity change', () => {
+      fixture.detectChanges();
+      const mockItem0Qty = mockItem0.quantity;
+      mockItem0.quantity = 20;
+      component.items = [mockItem0, mockItem1];
+      expect(
+        component.form.controls[mockItem0.entryNumber].get('quantity')?.value
+      ).toEqual(20);
+      mockItem0.quantity = mockItem0Qty;
```
Comment (bgambocjaviniar) [inline feature-libs/cart/base/components/cart-shared/cart-item-list/cart-item-list.component.spec.ts:308]: 👍  I understand we want to set back the quantity to the same it was before to not cause any leak for other future assertions for people, but first I need clarify[ this question before approving](https://github.com/SAP/spartacus/pull/15314/files#r824142667) (we might even remove this if you agree)

File: feature-libs/cart/base/components/cart-shared/cart-item-list/cart-item-list.component.spec.ts [hunk 3]
```diff
@@ -297,6 +297,17 @@ describe('CartItemListComponent', () => {
       ).toBeDefined();
     });
 
+    it('should update controls when quantity change', () => {
```
Comment (bgambocjaviniar) [inline feature-libs/cart/base/components/cart-shared/cart-item-list/cart-item-list.component.spec.ts:300]: I believe this should be 'existing', but we might remove depending on the answer in https://github.com/SAP/spartacus/pull/15314/files#r824142667
```suggestion
    it('should update existing controls when quantity change', () => {
```

**Review by bgambocjaviniar — CHANGES_REQUESTED** [review]
running all e2es* just to verify if saved cart got fixed for being able to focus and other e2es working as intended

Will revert previous commit once pipeline is done, but also change may be required depending on the answer related to https://github.com/SAP/spartacus/pull/15314/files#r824142667

EDIT: 

all e2es looking good 👍

File: feature-libs/cart/base/components/cart-shared/cart-item-list/cart-item-list.component.ts (hunk 1 above)
Comment (WeizhengSap) [reply feature-libs/cart/base/components/cart-shared/cart-item-list/cart-item-list.component.ts:174]: we use `setControl` because of this ticket https://github.com/SAP/spartacus/issues/14955.  We have to set back the qty for the case that the counter greater than the limit.

Comment (bgambocjaviniar) [reply feature-libs/cart/base/components/cart-shared/cart-item-list/cart-item-list.component.ts:174]: quick call during the day as I need to verify something

**Review by bgambocjaviniar — APPROVED** [review]
